File: Language/C#/Book/tmp/LINQ实战/Chapter13.CSharp - LinqBooks/LinqBooks.Web/Controls/BookList.ascx.cs
#region Imports

using System;
using System.Data;
using System.Configuration;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;

using LinqBooks.Entities;

#endregion Imports

namespace LinqBooks.Web.Controls
{
  public partial class BookList : System.Web.UI.UserControl
  {
    private IEnumerable<Book> _Books;

    public IEnumerable<Book> Books
    {
      set { _Books = value; }
    }

    protected void Page_Load(object sender, EventArgs e)
    {

    }

    public override void DataBind()
    {
      base.DataBind();
      GridViewBooks.DataSource = _Books;
      GridViewBooks.DataBind();
    }
  }
}

File: Language/C#/Book/tmp/LINQ实战/Chapter13.VB - LinqBooks/LinqBooks.Web/Controls/BookList.ascx.vb
Imports LinqBooks.Entities

Partial Public Class BookList
  Inherits System.Web.UI.UserControl

  Private _Books As IEnumerable(Of Book)

  Public WriteOnly Property Books() As IEnumerable(Of Book)
    Set(ByVal value As IEnumerable(Of Book))
      _Books = value
    End Set
  End Property

  Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load

  End Sub

  Public Overrides Sub DataBind()
    MyBase.DataBind()
    GridViewBooks.DataSource = _Books
    GridViewBooks.DataBind()
  End Sub
End Class